### PR TITLE
Add `Window::set_content_protected` on macOS and Windows

### PR DESCRIPTION
**Pull request by amrbashir**
- [x] Tested on all platforms changed
	- [x] macOS
	- [X] Windows
- [X] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [X] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [X] Created or updated an example program if it would help users understand this functionality
- [X] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

Porting https://github.com/tauri-apps/tao/commit/802146fb8692a46185846a64163c174520450c43

### CODE REVIEW

File: src/window.rs [hunk 1]
```diff
@@ -952,6 +952,18 @@ impl Window {
     pub fn theme(&self) -> Option<Theme> {
         self.window.theme()
     }
+
+    /// Prevents the window contents from being captured by other apps.
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **iOS / Android / x11 / Wayland / Web:** Unsupported.
+    pub fn set_content_protected(&self, enabled: bool) {
```
Comment (kchibisov) [inline src/window.rs:961]: Could name parameter with `_enabled`?

**Review by kchibisov — CHANGES_REQUESTED** [review]
You might want to have builder for that as well?

File: src/window.rs (hunk 1 above)
Comment (amrbashir) [reply src/window.rs:961]: right

**Review by msiglreith — APPROVED** [review]

**Review by madsmtm — COMMENTED** [review]
macOS impls looks good, unsure of how to test it though?

I tried recording the window with QuickTime, but that didn't seem to make a difference?

**Comment by amrbashir** [comment]
Looks like `NSWindowSharingNone` doesn't completely prevent all apps from reading the window content https://github.com/electron/electron/issues/14415#issuecomment-419736562
> This is OS API limitation. We use `[NSWindow setSharingType:NSWindowSharingNone]`, which does not prevent QuickTime from capturing the window unfortunately.

File: src/window.rs [hunk 2]
```diff
@@ -953,6 +968,16 @@ impl Window {
         self.window.theme()
     }
 
+    /// Prevents the window contents from being captured by other apps.
+    ///
```
Comment (madsmtm) [inline src/window.rs:972]: > > This is OS API limitation. We use `[NSWindow setSharingType:NSWindowSharingNone]`, which does not prevent QuickTime from capturing the window unfortunately.

Fair enough - would you mind documenting that here as well?

**Review by madsmtm — APPROVED** [review]
Thanks, macOS impl looks good

File: src/window.rs (hunk 2 above)
Comment (amrbashir) [reply src/window.rs:972]: sure

**Pull request merged**